From 0b7e99edcebaa5d6ebc5cbeec9e5446a4575b18f Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sun, 2 Nov 2014 21:00:32 -0500
Subject: [pci] Rename sysfs 'enabled' file back to 'enable'

Message-id: <20141102210032.14625.442.stgit@amt.stowe>
Patchwork-id: 98668
O-Subject: [RHEL7.1 PATCH] PCI: Rename sysfs 'enabled' file back to 'enable'
Bugzilla: 1159655
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1159655
Upstream Status: d8e7d53a2fc14e0830ab728cb84ee19933d3ac8d

Signed-off-by: Jarod Wilson <jarod@redhat.com>

Conflicts:
  ./drivers/pci/pci-sysfs.c
    Hunk #1 and #2 - Due to RHEL7 not having commit 3c78bc61f5ef ("PCI:
    Whitespace cleanup").  Commit 3c78bc61f5ef was too pervasive to bring
    in at this stage so only the changes relevant to this patch were
    included.  The bulk of commit 3c78bc61f5ef should get brought in
    during the next release as a result of further PCI subsystem
    re-basing.

Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=8187526
Testing: Successfully tested by looking in sysfs for filename change.
  [root@hp-dl360gen8-01 ~]# uname -a
  Linux hp-dl360gen8-01.rhts.eng.bos.redhat.com
  3.10.0-196.el7.rhbz1159655.x86_64 #1 SMP Sun Nov 2 13:40:44 EST 2014 x86_64
  x86_64 x86_64 GNU/Linux
  [root@hp-dl360gen8-01 ~]# ls -al /sys/devices/pci0000:00/0000:00:00.0/
  total 0
  drwxr-xr-x.  3 root root    0 Nov  2 15:52 .
  drwxr-xr-x. 34 root root    0 Nov  2 15:52 ..
  -rw-r--r--.  1 root root 4096 Nov  2 15:56 broken_parity_status
  -r--r--r--.  1 root root 4096 Nov  2 15:56 class
  -rw-r--r--.  1 root root 4096 Nov  2 15:56 config
  -r--r--r--.  1 root root 4096 Nov  2 15:56 consistent_dma_mask_bits
  -rw-r--r--.  1 root root 4096 Nov  2 15:56 d3cold_allowed
  -r--r--r--.  1 root root 4096 Nov  2 15:56 device
  -r--r--r--.  1 root root 4096 Nov  2 15:56 dma_mask_bits
  -rw-r--r--.  1 root root 4096 Nov  2 15:56 enable
  -r--r--r--.  1 root root 4096 Nov  2 15:52 irq
  -r--r--r--.  1 root root 4096 Nov  2 15:56 local_cpulist
  -r--r--r--.  1 root root 4096 Nov  2 15:56 local_cpus
  -r--r--r--.  1 root root 4096 Nov  2 15:52 modalias
  -rw-r--r--.  1 root root 4096 Nov  2 15:56 msi_bus
  -r--r--r--.  1 root root 4096 Nov  2 15:56 numa_node
  drwxr-xr-x.  2 root root    0 Nov  2 15:52 power
  --w--w----.  1 root root 4096 Nov  2 15:56 remove
  --w--w----.  1 root root 4096 Nov  2 15:56 rescan
  -r--r--r--.  1 root root 4096 Nov  2 15:56 resource
  lrwxrwxrwx.  1 root root    0 Nov  2 15:52 subsystem -> ../../../bus/pci
  -r--r--r--.  1 root root 4096 Nov  2 15:56 subsystem_device
  -r--r--r--.  1 root root 4096 Nov  2 15:56 subsystem_vendor
  -rw-r--r--.  1 root root 4096 Nov  2 15:52 uevent
  -r--r--r--.  1 root root 4096 Nov  2 15:56 vendor

commit d8e7d53a2fc14e0830ab728cb84ee19933d3ac8d
Author: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date:   Thu Oct 30 09:30:28 2014 -0700

    PCI: Rename sysfs 'enabled' file back to 'enable'

    Back in commit 5136b2da770d ("PCI: convert bus code to use dev_groups"),
    I misstyped the 'enable' sysfs filename as 'enabled', which broke the
    userspace API.  This patch fixes that issue by renaming the file back.

    Fixes: 5136b2da770d ("PCI: convert bus code to use dev_groups")
    Reported-by: Jeff Epler <jepler@unpythonic.net>
    Tested-by: Jeff Epler <jepler@unpythonic.net>       # on v3.14-rt
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    CC: stable@vger.kernel.org  # 3.13

diff --git a/drivers/pci/pci-sysfs.c b/drivers/pci/pci-sysfs.c
index 7c24546..fef5057 100644
--- a/drivers/pci/pci-sysfs.c
+++ b/drivers/pci/pci-sysfs.c
@@ -186,9 +186,8 @@ static ssize_t modalias_show(struct device *dev, struct device_attribute *attr,
 }
 static DEVICE_ATTR_RO(modalias);
 
-static ssize_t enabled_store(struct device *dev,
-        struct device_attribute *attr, const char *buf,
-        size_t count)
+static ssize_t enable_store(struct device *dev, struct device_attribute *attr,
+        const char *buf, size_t count)
 {
  struct pci_dev *pdev = to_pci_dev(dev);
  unsigned long val;
@@ -212,15 +211,15 @@ static ssize_t enabled_store(struct device *dev,
  return result < 0 ? result : count;
 }
 
-static ssize_t enabled_show(struct device *dev,
-       struct device_attribute *attr, char *buf)
+static ssize_t enable_show(struct device *dev, struct device_attribute *attr,
+       char *buf)
 {
  struct pci_dev *pdev;
 
  pdev = to_pci_dev (dev);
  return sprintf (buf, "%u\n", atomic_read(&pdev->enable_cnt));
 }
-static DEVICE_ATTR_RW(enabled);
+static DEVICE_ATTR_RW(enable);
 
 #ifdef CONFIG_NUMA
 static ssize_t
@@ -526,7 +525,7 @@ static struct attribute *pci_dev_attrs[] = {
 #endif
  &dev_attr_dma_mask_bits.attr,
  &dev_attr_consistent_dma_mask_bits.attr,
- &dev_attr_enabled.attr,
+ &dev_attr_enable.attr,
  &dev_attr_broken_parity_status.attr,
  &dev_attr_msi_bus.attr,
 #if defined(CONFIG_PM_RUNTIME) && defined(CONFIG_ACPI)
-- 
1.7.1